stubs/setuptools/setuptools/_distutils/compat/__init__.pyi

```diff
@@ -0,0 +1 @@
+def consolidate_linker_args(args: list[str]) -> str: ...
```

Comment:
I think the type hint in the runtime source is incorrect here? It looks like if any of the strings in the input list don't start with `"-Wl,"`, then it returns a list of strings rather than joining them together?

https://github.com/pypa/setuptools/blob/ff58075cdf3459ecdf73486d2a83cecdd70c7e4a/setuptools/_distutils/compat/__init__.py#L13-L14

I don't know if that's... deliberate...

```suggestion
def consolidate_linker_args(args: list[str]) -> str | list[str]: ...
```

Reply:
Looks like a bug to me. 🤷‍♂️